pkg/sdk/reconciler/reconciler.go

```diff
@@ -544,7 +544,7 @@ func (r *Reconciler) GetAllDeployments(cr client.Object) ([]*appsv1.Deployment,
 // WatchCR registers watch for the managed CR
 func (r *Reconciler) WatchCR() error {
 	// Watch for changes to managed CR
-	return r.controller.Watch(source.Kind(r.getCache(), r.crManager.Create()), &handler.EnqueueRequestForObject{})
+	return r.controller.Watch(source.Kind(r.getCache(), r.crManager.Create(), &handler.EnqueueRequestForObject{}))
```

Comment:
was going to tag this and ask how we feel about not doing `TypedEnqueueRequestForObject` here

Reply:
It is perfectly fine because the type returned by `r.crManager.Create()` is client.Object

Reply:
yeah but couldn't we get rid of this in favor of `TypedEnqueueRequestForObject` to receive the benefits of https://github.com/kubernetes-sigs/controller-runtime/pull/2783

Reply:
I don't see much of an advantage to us.  What specifically are you keen on?  

If you feel strongly, it would require an API change (add template param to cr manager) and best handled in subsequent PR

Reply:
I guess the main thing would be something like avoiding a type assertion inside the predicate which we're anyway not using here, so I am fine